### PR TITLE
RD-5206 The unsaved filter in Service page cannot be reused if removed

### PR DESCRIPTION
**Pull request by PatrycjaKubiczek**
## Description

RD-5206
- remove useEffect which cleaned state thus the unsaved filters couldn't be reused

## Screenshots / Videos

Before:

https://user-images.githubusercontent.com/17828621/180976096-16711fa9-c30e-494c-acf7-bbf41c433269.mp4

After:

https://user-images.githubusercontent.com/17828621/180974407-74c32d32-5be6-44af-8070-07be7a13fd74.mp4

## Checklist

- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I verified that all tests and checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests

 

filters_spec: 
[Build Status](https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/3220/)

## Documentation

n/a

### CODE REVIEW

File: widgets/common/src/deploymentsView/header/filter/FilterModal.tsx [hunk 1]
```diff
@@ -46,13 +46,7 @@ function useRevertableStates(...revertableStates: Omit<ReturnType<typeof useReve
...
 }
 
-const FilterModal: FunctionComponent<FilterModalProps> = ({
-    userFilterSelected,
-    open,
-    onCancel,
-    onSubmit,
-    toolbox
-}) => {
+const FilterModal: FunctionComponent<FilterModalProps> = ({ open, onCancel, onSubmit, toolbox }) => {
```
Comment (NorbertAmartus) [inline widgets/common/src/deploymentsView/header/filter/FilterModal.tsx:49]: As we are removing `useFilterSelected` prop, we should also do a little cleanup on the [parent component](https://github.com/cloudify-cosmo/cloudify-stage/blob/master/widgets/common/src/deploymentsView/header/index.tsx#L93) side, to not pass this prop anymore (as it's no longer being used) 😉

**Review by NorbertAmartus — COMMENTED** [review]
Simple yet elegant solution 😁 

Left one comment regarding some code residues

File: widgets/common/src/deploymentsView/header/filter/FilterModal.tsx (hunk 1 above)
Comment (PatrycjaKubiczek) [reply widgets/common/src/deploymentsView/header/filter/FilterModal.tsx:48]: great catch 👍 removed unnecessary props: https://github.com/cloudify-cosmo/cloudify-stage/pull/2091/commits/b3cc76749409ea672aea45c7717e5785e5522329

**Review by qooban — APPROVED** [review]

**Comment by PatrycjaKubiczek** [comment]
@qooban should this go to 6.4 release?
cc @NorbertAmartus

**Comment by qooban** [comment]
@PatrycjaKubiczek Yes. Let's merge it to `master` and then create cherry-pick PR to `6.4.0-build` branch.

**Pull request merged**